lib/erlef/erlef/blog/post.ex

```diff
@@ -0,0 +1,29 @@
+defmodule Erlef.Blog.Post do
+  @moduledoc """
+  Schema for blog posts.
+  """
+  use Ecto.Schema
+  import Ecto.Changeset
+
+  schema "blog_posts" do
+    field(:body, :string)
+    field(:slug, :string)
+    field(:status, :string)
```

Comment:
What do you think about making status an enum (ecto enum) and checking the status in the changeset vs on the db side? Rationale is 1. the application can use atoms for the status 2. Since performance isn't a concern, we can gain flexibility by only having to modify code in the case of adding or removing valid statuses?

Reply:
Great suggestion! Will definitely go with `Ecto.Enum` here, and drop the constraint-based approach. Now that I think of it, the definition of different statues is definitely a business-side requirement anyway, not something the DB should care this much about.